megamek/src/megamek/common/actions/ArtilleryAttackAction.java

```diff
@@ -49,26 +50,41 @@ public ArtilleryAttackAction(int entityId, int targetType, int targetId,
         this.launchVelocity = 50;
         int distance = Compute.effectiveDistance(game, getEntity(game),
                 getTarget(game));
-        // adjust distance for gravity
+        //adjust distance for gravity
         distance = (int)Math.floor((double)distance/game.getPlanetaryConditions().getGravity());
         EquipmentType eType = game.getEntity(entityId).getEquipment(weaponId).getType();
         WeaponType wType = (WeaponType) eType;
+        if (game.getEntity(entityId).usesWeaponBays()) {
+            for (int wId : game.getEntity(entityId).getEquipment(weaponId).getBayWeapons()) {
+                Mounted bayW = game.getEntity(entityId).getEquipment(wId);
+                WeaponType bayWType = ((WeaponType) bayW.getType());
+                if (bayWType.hasFlag(WeaponType.F_CRUISE_MISSILE)) {
+                    break;
+                }
+            }
+            turnsTilHit = 1 + (distance / 17 / 5);
```

Comment:
Magic numbers, magic strings, they're all over the place.  I'm betting that the 17 is a reference to the board length.  As I recall, arty typically requires 1 extra turn per board length or something like that.  Not sure on the 5.

Reply:
Cruise missile flight time is 1 + (Number of mapsheets ÷ 5, round down) (TO p 181)

Reply:
Oh, fine, look things up and take all the fun out of guesswork, why don't you. ;-P

Reply:
So what's the deal here?  Do you want me to go through the whole class and clean it up.  I could see 17 being a named constant since board length applies lots of places.  I don't think the 5 should be.  That whole line could be explained with a comment.    There seems to be a difference of opinion here, so I'd like to know which way I'm going with it...